### PR TITLE
dfelibs: new package

### PR DESCRIPTION
**Pull request by stephenswat**
This commits adds the `dfelibs` package, which is an Acts project fork of dfelibs.

### CODE REVIEW

File: var/spack/repos/builtin/packages/acts-dfelibs/package.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+class ActsDfelibs(CMakePackage):
+    """Acts fork of dfelibs, a header-only untility library for C++."""
+
+    homepage = "https://github.com/acts-project/dfelibs"
+    url = "https://github.com/acts-project/dfelibs/archive/refs/tags/v20211029.tar.gz"
+    list_url = "https://github.com/acts-project/dfelibs/tags"
+
+    maintainers("stephenswat")
+
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/acts-dfelibs/package.py]:
```suggestion

    license("MIT", checked_by="stephenswat")

```

File: var/spack/repos/builtin/packages/acts-dfelibs/package.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class ActsDfelibs(CMakePackage):
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/acts-dfelibs/package.py]:
```suggestion
class Dfelibs(CMakePackage):
```

We typically avoid naming packages differently for forks than for their upstream. I understand that upstream is unmaintained, but this avoids that someone else creates another `dfelibs` package independently of this one.

If at some point someone wants to add upstream (and cannot use the acts fork), or another fork, and if there really are conflicts between the patch sets that prevent merging into acts as a de facto maintained version, then we can always figure out something then.

File: var/spack/repos/builtin/packages/acts-dfelibs/package.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    """Acts fork of dfelibs, a header-only untility library for C++."""
+
+    homepage = "https://github.com/acts-project/dfelibs"
+    url = "https://github.com/acts-project/dfelibs/archive/refs/tags/v20211029.tar.gz"
+    list_url = "https://github.com/acts-project/dfelibs/tags"
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/acts-dfelibs/package.py]: I'm curious: is this list_url necessary (again)?

Spack should be able to deduce the list URL for GitHub tag archives automatically, but occasionally GitHub changes something (including hiding html content behind dynamically loading JavaScript calls...) that breaks what we do.

Comment (stephenswat) [reply var/spack/repos/builtin/packages/acts-dfelibs/package.py]: I guess not; I will remove it and test it.

**Comment by stephenswat** [comment]
Thank you for the feedback @wdconinc; I have removed the `acts` from the name, added you as a maintainer, updated the license, and removed the unnecessary URL variable. I tested that the spec still installs! :+1:

File: var/spack/repos/builtin/packages/dfelibs/package.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Dfelibs(CMakePackage):
+    """Acts fork of dfelibs, a header-only untility library for C++."""
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/dfelibs/package.py]:
```suggestion
    """Acts fork of dfelibs, a header-only utility library for C++."""
```

**Review by wdconinc — APPROVED** [review]
Thanks for adding this!

**Pull request merged**